### ISSUE TITLE
Apply inverted '--path' filtering after '--path-rename'?

### ISSUE THREAD

**Issue by MarkusMit** [newren/git-filter-repo#504]
I have a repository like this:
```
my-repo/
├── folder_a/
├── folder_b/
│   ├── folder_b_1/
│   ├── folder_b_2/
│   └── folder_b_3/
├── folder_c/
...
└── folder_z/
```
and - as part of a bigger restructuring - I want to change it like this: 
`folder_b_3` should be moved up to root, and folder `folder_b` with all its remaining contents should be removed:
```
my-repo/
├── folder_a/
├── folder_b_3/
├── folder_c/
...
└── folder_z/
```

I tried to do this with inverted path filtering (`--invert-paths`), but it seems the `--path` filtering is always executed before the `--path-rename` regardless of the order of the commands:
```bash
python3 'git-filter-repo' --force --invert-paths              \
            --path-rename 'folder_b/folder_b_3/:folder_b_3/'  \
            --path 'folder_b/' 
```

So, the current (undesired) result is, that nothing of `folder_b` survives:
```
my-repo/
├── folder_a/
├── folder_c/
...
└── folder_z/
```

Since "ordering issues" are mentioned with (positive) path filtering and renames in the [user manual](https://htmlpreview.github.io/?https://github.com/newren/git-filter-repo/blob/docs/html/git-filter-repo.html), I thought it should also be considered accordingly with inverted path filtering.

**Comment by newren** [comment]
Yeah, sorry, can you split this into two steps?
```
   python3 'git-filter-repo' --force \
            --path-rename 'folder_b/folder_b_3/:folder_b_3/'
   python3 'git-filter-repo' --invert-paths \
            --path 'folder_b/' 
```

**Comment by MarkusMit** [comment]
Yeah, since this was already some months ago, I already found a solution for it. ;-)

But my main intend was anyway that this behaviour may be worth mentioning in the documentation.

**Comment by newren** [comment]
Sorry for taking so long to fix this. I looked at it a time or two in the past and was perplexed. Had a facepalm moment today.

Turns out this did work once upon a time, but was broken by 5c4637f (Documentation: add guides for people converting from filter-branch or BFG, 2020-05-16). Anyway, now fixed by 19b1cb6 (filter-repo: also allow --path-rename arguments before --path arguments, 2024-11-25) which I just pushed.

(Also, to make matters worse, this is a duplicate of #142, meaning multiple people hit this issue.)
